zio-http/src/main/scala/zhttp/socket/SocketProtocol.scala

```diff
@@ -1,19 +1,29 @@
 package zhttp.socket
 
-import io.netty.handler.codec.http.websocketx.{WebSocketCloseStatus, WebSocketServerProtocolConfig}
+import io.netty.handler.codec.http.websocketx.{
+  WebSocketClientProtocolConfig,
+  WebSocketCloseStatus,
+  WebSocketServerProtocolConfig,
+}
+import zhttp.http.Headers
+import zhttp.socket.SocketProtocol.Config.{ClientConfig, ServerConfig}
 import zio.duration.Duration
 
 /**
  * Server side websocket configuration
  */
-sealed trait SocketProtocol { self =>
+sealed trait SocketProtocol[+A] { self =>
   import SocketProtocol._
-  def ++(other: SocketProtocol): SocketProtocol = SocketProtocol.Concat(self, other)
-  def javaConfig: WebSocketServerProtocolConfig = {
+  def narrow[A1]: SocketProtocol[A1] = self.asInstanceOf[SocketProtocol[A1]]
```

Comment:
Do we need this?

Reply:
I was trying something with types, but couldn't make it work. My idea was to enforce subtyping between `A1` and `A` to eliminate the use of `asInstanceOf` without type checks.